pddl/parser/base.py

```diff
@@ -0,0 +1,70 @@
+#
+# Copyright 2021-2025 WhiteMech
+#
+# ------------------------------
+#
+# This file is part of pddl.
+#
+# Use of this source code is governed by an MIT-style
+# license that can be found in the LICENSE file or at
+# https://opensource.org/licenses/MIT.
+#
+
+"""Base classes for PDDL parsers."""
+import sys
+from abc import ABC
+from typing import Any, Type, TypeVar
+
+from lark import Lark, Transformer
+
+from pddl.parser import GRAMMAR_FILE, PARSERS_DIRECTORY
+
+T = TypeVar("T")
+
+
+class BaseParser(Transformer[Any, T], ABC):
```

Comment:
Not extremely necessary to add `BaseParser` for this PR, but it might be useful later when we want to refactor the parsing APIs.